oak_runtime/src/tests.rs

```diff
@@ -49,9 +50,18 @@ fn run_node_body(node_label: &Label, node_privilege: &NodePrivilege, node_body:
         &signature_table,
     );
 
+    // Result of a test, shared between the main test thread and the per-Node thread.
+    struct TestResult {
+        // Result of executing a `NodeBody` once complete (`None` before then).
+        value: Mutex<Option<Result<(), OakStatus>>>,
+        // Condition variable used to signal the completion of the test.
+        cv: Condvar,
+    };
+
     struct TestNode {
         node_body: Box<NodeBody>,
         node_privilege: NodePrivilege,
+        result: Arc<TestResult>,
```

Comment:
Have you considered a https://doc.rust-lang.org/std/sync/mpsc/fn.sync_channel.html ?

Reply:
A condition variable leapt to mind because this situation is exactly what they're designed for, but the Rust `Condvar` is kind of fiddly compared to some other implementations (e.g. Abseil's [`Mutex` with built-in condition var](https://abseil.io/docs/cpp/guides/synchronization#conditional-critical-sections)). 

`sync_channel` looks much cleaner (even though we don't need a lot of its function) so I'll give it a go.

Reply:
Yep, that works and makes the code much easier to follow than the naked `Condvar` version &ndash; thanks!

It also seems to have solved the problem with the previous version where a failed assertion in a test would result in the test hanging, but I don't yet understand why&hellip;

Reply:
(Answer: `mpsc:sender_receiver` detects if the sender has disconnected, including due to thread death, and emits a `RecvError` from the `receiver.recv()` call.)

Reply:
yeah that was my mental process too: "nice, I guess CondVar is meant for exactly this case it seems... if only there was something that keeps a condvar together with the inner value.. I guess that would be a sort of channel... and it may also detect when the sender disappears without sending anything" :)